src/renderers/shared/fiber/ReactFiberReconciler.js

```diff
@@ -182,7 +185,14 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
       }
     }
 
-    const priorityLevel = getPriorityContext();
+    // Check if the top-level element is an async wrapper component. If so, treat
+    // updates to the root as async. This is a bit weird but lets us avoid a separate
+    // `renderAsync` API.
+    const forceAsync = ReactFeatureFlags.enableAsyncSubtreeAPI &&
+      element != null &&
+      element.type != null &&
+      (element.type: any).unstable_asyncUpdates === true;
+    const priorityLevel = getPriorityContext(current, forceAsync);
```

Comment:
and I only want to override the result of `getPriorityContext` if it's the default priority.

We could also solve by getting rid of `getPriorityContext` and moving `addUpdate`, `addTopLevelUpdate`, et al back into the scheduler, but we previously decided to move them out because it the scheduler was getting too bloated.